dashboard/app/models/lesson.rb

```diff
@@ -442,18 +442,23 @@ def lockable_state(students)
     end
     return students.map do |student|
       user_level = student.last_attempt_for_any script_level.levels, script_id: script.id
-      # user_level_data is provided so that we can get back to our user_level when updating. in some cases we
-      # don't yet have a user_level, and need to provide enough data to create one
+      # user_level_data is provided so that we can get back to our user_level
+      # when updating. in some cases we don't yet have a user_level, and need
+      # to provide enough data to create one
+
+      # if we don't have a user level, consider ourselves locked
+      locked = user_level.nil? || user_level.show_as_locked?(self)
+      # if we don't have a user level, we can't be readonly
```

Comment:
Why is this true?  (I'm not sure I understand what it means to not have a `user_level`.)

Reply:
(discussed offline)- user_level is created when an assessment is first unlocked)